engines/scumm/string.cpp

```diff
@@ -454,6 +461,88 @@ bool ScummEngine::newLine() {
 	return true;
 }
 
+void ScummEngine::fakeBidiString(char *ltext, bool ignoreVerb) {
+	int ll = 0;
+	if (_game.id == GID_INDY4 && ltext[ll] == 127) {
+		ll++;
+	}
+	while (ltext[ll] == -1) {
+		ll += 4;
+	}
+	int ipos = 0;
+	int start = 0;
+	char* text = ltext + ll;
+	char* current = text;
+	while(1) {
+		if (*current == 13 || *current == 0 || *current == -1 || *current == -2) {
+
+			// ignore the line break for verbs texts
+			if (ignoreVerb && (*(current + 1) ==  8)) {
+				*(current + 1) = *current;
+				*current = 8;
+				ipos += 2;
+				current += 2;
+				continue;
+			}
+
+			char buff[384];
```

Comment:
These are quite big for the stack. I recommend to `malloc` these two buffers  at the method start and free on the exit.